### PR TITLE
Service question model

### PR DESCRIPTION
**Pull request by naegahyung**
Java data models for questions to a service, and answers to a service are added

### CODE REVIEW

File: src/main/java/edu/neu/cs4500/models/Answer.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
+package edu.neu.cs4500.models;
+
+import javax.persistence.*;
+import java.util.Date;
+
+public class Answer {
+    @Id
+    @GeneratedValue(strategy= GenerationType.IDENTITY)
+    private Integer id;
+    private Object body;
```
Comment (adamtobey) [inline src/main/java/edu/neu/cs4500/models/Answer.java]: I think body should be a String.

File: src/main/java/edu/neu/cs4500/models/Answer.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
+package edu.neu.cs4500.models;
+
+import javax.persistence.*;
+import java.util.Date;
+
+public class Answer {
```
Comment (adamtobey) [inline src/main/java/edu/neu/cs4500/models/Answer.java]: Maybe rename to ServiceQuestionAnswer, and link the table `service_question_answers`

File: src/main/java/edu/neu/cs4500/models/Question.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
+package edu.neu.cs4500.models;
+
+import javax.persistence.*;
+import java.util.Date;
+
+@Entity
+@Table(name="questions")
+public class Question {
```
Comment (adamtobey) [inline src/main/java/edu/neu/cs4500/models/Question.java]: Maybe rename to ServiceQuestion

File: src/main/java/edu/neu/cs4500/models/Answer.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    @Id
+    @GeneratedValue(strategy= GenerationType.IDENTITY)
+    private Integer id;
+    private Object body;
+    private User updatedBy;
+
```
Comment (adamtobey) [inline src/main/java/edu/neu/cs4500/models/Answer.java]: Should have a link to the Service as a foreign key

File: src/main/java/edu/neu/cs4500/models/Service.java [hunk 5]
```diff
@@ -13,6 +9,18 @@
 	@GeneratedValue(strategy=GenerationType.IDENTITY)
 	private Integer id;
 	private String serviceName;
+
+	@OneToMany(cascade = CascadeType.REMOVE, orphanRemoval = true)
+	private Question questions;
```
Comment (adamtobey) [inline src/main/java/edu/neu/cs4500/models/Service.java]: Shouldn't this be a `List<Question>`?

File: src/main/java/edu/neu/cs4500/models/Service.java [hunk 6]
```diff
@@ -13,6 +9,18 @@
...
+
+	public Question getQuestions() {
+		return questions;
+	}
+
+	public void setQuestions(Question questions) {
```
Comment (adamtobey) [inline src/main/java/edu/neu/cs4500/models/Service.java]: We can't support this method, let's add questions by creating a question with a fkey to this service

File: src/main/java/edu/neu/cs4500/models/Type.java [hunk 7]
```diff
@@ -0,0 +1,7 @@
+package edu.neu.cs4500.models;
+
+public enum Type {
```
Comment (adamtobey) [inline src/main/java/edu/neu/cs4500/models/Type.java]: Let's rename to ServiceQuestionType

File: src/main/java/edu/neu/cs4500/models/Answer.java (hunk 4 above)
Comment (naegahyung) [reply src/main/java/edu/neu/cs4500/models/Answer.java]: what do you mean by this?

File: src/main/java/edu/neu/cs4500/models/Service.java (hunk 6 above)
Comment (naegahyung) [reply src/main/java/edu/neu/cs4500/models/Service.java]: Question model has ManyToOne mapping from Question, where we can set service.

**Comment by mmzzyh** [comment]
LGTM now.

**Pull request merged**